django/db/models/fields/related.py

```diff
@@ -854,6 +856,59 @@ def _check_on_delete(self):
                     id='fields.E321',
                 )
             ]
+        elif on_delete == DB_CASCADE and (
+            any(  # generic relation
+                hasattr(field, 'bulk_related_objects') for field in
+                self.model._meta.private_fields
+            ) or
+            any(  # generic foreign key
+                hasattr(field, 'get_content_type') for field in
+                self.model._meta.private_fields
+            )
+        ):
+            return [
+                checks.Warning(
+                    'Field specifies unsupported DB_CASCADE on model '
+                    'declaring a GenericForeignKey.',
+                    hint='Change the on_delete rule.',
+                    obj=self,
+                    id='fields.W346'
+                )
+            ]
+        elif on_delete == DB_CASCADE and len(
+            # multi table inheritance
+            self.model._meta.concrete_model._meta.parents.values()
+        ):
+            return [
+                checks.Warning(
+                    'Field specifies unsupported DB_CASCADE on Multi-table '
```

Comment:
If these checks are all for "unsupported" behaviour, should they not be `Error` instead of `Warning`?